optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -183,12 +183,10 @@ public final void solve(Solution planningProblem) {
     public void outerSolvingStarted(DefaultSolverScope solverScope) {
         solving.set(true);
         basicPlumbingTermination.resetTerminateEarly();
-        solverScope.setStartingSystemTimeMillis(System.currentTimeMillis());
```

Comment:
Yes the startingSovlerCount should be placed in the outerSolverStarted. Agreed that the outerSolverStarting time could be useful. Because in multi-tenance application, each daemon mode solver will hog up a cpu core and therefore is resource consuming (more $$ charging for tenants). Some tenants may want to kill the always-on solver after the operation hours (e.g. 9am to 9pm for a vehicle routing application). Then the optaplanner execution server cluster can serve more customers with less cpu cores with the help of a job dispatching layer. The outerSolvingTime can be used to indicate how long the "solving application" has been running and the solver can stop restarting the solver in daemon once the time termination config has been met. private boolean checkProblemFactChanges(){	if (termination.isDaemonModeTerminated(System.currentTimeMillis - outerSolverStaringTime){		return false;	}…..…..}private boolean checkProblemFactChanges() {                             Original Message                             Sender: Geoffrey De Smet<notifications@github.com>Recipient: droolsjbpm/optaplanner<optaplanner@noreply.github.com>Cc: isaaczhangadventurer<18821688@qq.com>Date: Tuesday, Jun 2, 2015 17:41Subject: Re: [optaplanner] Daemon Mode Bug fixed  (#89)In optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java:
> @@ -183,12 +183,10 @@ public final void solve(Solution planningProblem) {
>      public void outerSolvingStarted(DefaultSolverScope solverScope) {
>          solving.set(true);
>          basicPlumbingTermination.resetTerminateEarly();
> -        solverScope.setStartingSystemTimeMillis(System.currentTimeMillis());

Agreed that resets are need, for time terminations, for timegradient, etc.
But don't we need to track the outerStarting time too, for some functionality? Investigating now.

—Reply to this email directly or view it on GitHub.